scripts/README.md

```diff
@@ -0,0 +1,43 @@
+# Scripts
+
+This directory contains numerous short C# scripts, each in their own directory
+with its own project file.
+
+To create a new script, copy an existing script directory, and:
+
+* Rename the csproj file to match the directory name.
+* Add your C# code.
+* Edit the arguments to the `TemplateScript` template in the `fragment.mk`
+  file according to how you named your script (directory). The first argument
+  will be used in other makefiles that use the script, the second is the name
+  of the script (directory). Say your script is `my-script`, then that would be:
```

Comment:
would it make sense to write all of this into the Makefile such that I could do something like this:
```bash
make scriptname=MyNewScript new-script
```
And this will:
1 - make the directory
2 - copy a templated version of fragment.mk and sed in the new name or some other templating tool
3 - remind you of next steps

For each script, I'd like to see an individual README.md to document at a minimum what the script does or its intended use.

Reply:
@stephen-hawley I added a shell script to create a new script, it seemed easier to do it that way than in `make`.

I also documented all the other scripts, so this is ready for a re-review now.